### ISSUE TITLE
Triple stream interrupted before it's consumed

### ISSUE THREAD

**Issue by Robsteranium** [rubensworks/fetch-sparql-endpoint.js#40]
I'm trying to do a construct query in the browser:

```js
import { SparqlEndpointFetcher } from "fetch-sparql-endpoint";

const fetcher = new SparqlEndpointFetcher();
const endpoint = 'https://dbpedia.org/sparql';
const query = 'CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o } LIMIT 5000'

async function queryConstruct(fetcher, endpoint, query) {
  const tripleStream = await fetcher.fetchTriples(endpoint, query)

  let data = []
  tripleStream
    .on('data', (d) => data.push(d))
    .on('end', () => console.log(`Read ${data.length} items`))
}

queryConstruct(fetcher, endpoint, query);
```

I'm getting the following intermittent error when I have high LIMITs...

In Firefox:

> Uncaught (in promise) TypeError: The ReadableStream reader method 'read' may only be called on a reader owned by a stream.

In Chromium:

> Uncaught (in promise) TypeError: Failed to execute 'read' on 'ReadableStreamDefaultReader': This readable stream reader has been released and cannot be used to read from its previous owner stream

This doesn't happen running similar code in node. I'm using webpack 5.51.1 which means I've had [explicitly polyfill node core modules](https://webpack.js.org/configuration/resolve/#resolvefallback). My `webpack.config.js`:

```js
const webpack = require('webpack')

module.exports = {
  mode: 'development',
  resolve: {
    fallback: {
      stream: require.resolve("stream-browserify"),
      buffer: require.resolve("buffer"),
      process: require.resolve("process/browser"),
      util: require.resolve("util")
    },
  },
  plugins: [  
    new webpack.ProvidePlugin({
      process: 'process/browser',
    }),
    new webpack.ProvidePlugin({
      Buffer: ['buffer', 'Buffer'],
    })
  ]
};
```

I've tried using both the `readable-stream` and `stream-browserify` replacements for node.js `stream`.

I've also tried using browserify instead of webpack.

Stacktraces...

<details>
  <summary>Firefox</summary>
<pre>
Uncaught (in promise) TypeError: The ReadableStream reader method 'read' may only be called on a reader owned by a stream.
    doRead web-streams-node/lib/conversions.js?:45
    _read web-streams-node/lib/conversions.js?:66
    read readable-stream/lib/_stream_readable.js?:457
    flow readable-stream/lib/_stream_readable.js?:940
    pipeOnDrainFunctionResult readable-stream/lib/_stream_readable.js?:750
    emit events/events.js?:153
    onwriteDrain readable-stream/lib/_stream_writable.js?:480
    afterWrite readable-stream/lib/_stream_writable.js?:468
    run process/browser.js?:153
    drainQueue process/browser.js?:123
</pre>
</details>

<details>
  <summary>Chromium</summary>
<pre>
conversions.js:45 Uncaught (in promise) TypeError: Failed to execute 'read' on 'ReadableStreamDefaultReader': This readable stream reader has been released and cannot be used to read from its previous owner stream
    at doRead (conversions.js:45)
    at NodeReadable._read (conversions.js:66)
    at NodeReadable.Readable.read (_stream_readable.js:457)
    at flow (_stream_readable.js:940)
    at N3StreamParser.pipeOnDrainFunctionResult (_stream_readable.js:750)
    at N3StreamParser.emit (events.js:153)
    at onwriteDrain (_stream_writable.js:480)
    at afterWrite (_stream_writable.js:468)
    at Item.run (browser.js:153)
    at drainQueue (browser.js:123)
</pre>
</details>

Any suggestions on how to get this working?

I'm not sure what the interruption is... back pressure?

**Comment by rubensworks** [comment]
I can reproduce this at http://query.linkeddatafragments.org/#datasources=https%3A%2F%2Fdbpedia.org%2Fsparql&query=CONSTRUCT%20%7B%20%3Fs%20%3Fp%20%3Fo%20%7D%20WHERE%20%7B%20%3Fs%20%3Fp%20%3Fo%20%7D%20LIMIT%205000

Not sure what is going on though, as it looks like this error does _not_ occur when running this Web client in development-mode...

But I do suspect that it has something to do with the conversion of a fetch stream to a node stream: https://github.com/rubensworks/fetch-sparql-endpoint.js/blob/master/lib/SparqlEndpointFetcher.ts#L211-L212
It seems like the library we use for this has been deprecated: https://github.com/JCCR/web-streams-node

It looks like Node 16 has experimental support for a new `Readable.fromWeb` function though: https://github.com/nodejs/node/blob/master/doc/api/stream.md#streamreadablefromwebreadablestream-options

Unsure about any short-term solutions...

**Comment by rubensworks** [comment]
Fixed in 2.3.2

**Comment by Robsteranium** [comment]
Thanks!
